components/dashboard/src/components/menu.tsx

```diff
@@ -4,201 +4,249 @@
+import AppBar from "@material-ui/core/AppBar";
 import Avatar from "@material-ui/core/Avatar";
-import Button from '@material-ui/core/Button';
-import Toolbar from '@material-ui/core/Toolbar';
-import Menu from '@material-ui/core/Menu';
-import MenuItem from '@material-ui/core/MenuItem';
-import Fade from '@material-ui/core/Fade';
-import IconButton from '@material-ui/core/IconButton';
-import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
-import * as React from 'react';
-import { ButtonWithMenu } from './button-with-menu';
-import { getSvgPath } from '../withRoot';
+import Button from "@material-ui/core/Button";
+import Toolbar from "@material-ui/core/Toolbar";
+import Menu from "@material-ui/core/Menu";
+import MenuItem from "@material-ui/core/MenuItem";
+import Fade from "@material-ui/core/Fade";
+import IconButton from "@material-ui/core/IconButton";
+import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
+import * as React from "react";
+import { ButtonWithMenu } from "./button-with-menu";
+import { getSvgPath } from "../withRoot";
+import MobileMenu from "./mobile-menu";
 
 export interface MainMenuProps {
-    menuEntries: MenuEntry[];
+  menuEntries: MenuEntry[];
 
-    user?: User;
-    authProviders: AuthProviderInfo[];
-    userLoaded: boolean;
-    branding?: Branding;
+  user?: User;
+  authProviders: AuthProviderInfo[];
+  userLoaded: boolean;
+  branding?: Branding;
 
-    linksOverride?: JSX.Element;
+  linksOverride?: JSX.Element;
 }
 
 interface MainMenuState {
-    anchorElement: HTMLElement | undefined;
+  anchorElement: HTMLElement | undefined;
 }
 
 export interface MenuEntry {
-    caption: string,
-    open: (loc: Location) => string,
+  caption: string;
+  open: (loc: Location) => string;
 }
 
 export namespace MenuEntry {
-    export const ACCESS_CONTROL: MenuEntry = {
-        caption: 'Access Control',
-        open: (loc) => new GitpodHostUrl(loc.toString()).asAccessControl().toString()
-    };
-    export const SETTINGS: MenuEntry = {
-        caption: 'Settings',
-        open: (loc) => new GitpodHostUrl(loc.toString()).asSettings().toString()
-    };
-    export const LOG_OUT: MenuEntry = {
-        caption: 'Log Out',
-        open: (loc) => new GitpodHostUrl(loc.toString()).withApi({ pathname: '/logout' }).toString()
-    };
-    export const ALL: MenuEntry[] = [ACCESS_CONTROL, SETTINGS, LOG_OUT];
+  export const ACCESS_CONTROL: MenuEntry = {
+    caption: "Access Control",
+    open: (loc) =>
+      new GitpodHostUrl(loc.toString()).asAccessControl().toString(),
+  };
+  export const SETTINGS: MenuEntry = {
+    caption: "Settings",
+    open: (loc) => new GitpodHostUrl(loc.toString()).asSettings().toString(),
+  };
+  export const LOG_OUT: MenuEntry = {
+    caption: "Log Out",
+    open: (loc) =>
+      new GitpodHostUrl(loc.toString())
+        .withApi({ pathname: "/logout" })
+        .toString(),
+  };
+  export const ALL: MenuEntry[] = [ACCESS_CONTROL, SETTINGS, LOG_OUT];
```

Comment:
To make reviewing & iterating easier, please don't reformat entire files.

E.g. here, the indentation seemed to be 4 spaces here. Please continue using 4-space indentation in this file / directory. This will minimize the diff noise; make reviewing easier by highlighting only what actually changed; and minimize the risk of conflicts on the other re-indented lines.